src/main/java/uk/gov/companieshouse/acsp/manage/users/service/AcspMembersService.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.companieshouse.acsp.manage.users.service;
+
+import java.util.List;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.companieshouse.acsp.manage.users.mapper.AcspMembershipListMapper;
+import uk.gov.companieshouse.acsp.manage.users.model.UserContext;
+import uk.gov.companieshouse.acsp.manage.users.repositories.AcspMembersRepository;
+import uk.gov.companieshouse.api.acsp_manage_users.model.AcspMembership;
+
+@Service
+public class AcspMembersService {
+
+  private final AcspMembersRepository acspMembersRepository;
+  private final AcspMembershipListMapper acspMembershipListMapper;
+
+  public AcspMembersService(
+      final AcspMembersRepository acspMembersRepository,
+      final AcspMembershipListMapper acspMembershipListMapper) {
+    this.acspMembersRepository = acspMembersRepository;
+    this.acspMembershipListMapper = acspMembershipListMapper;
+  }
+
+  @Transactional(readOnly = true)
+  public List<AcspMembership> fetchAcspMemberships(
+      final String userId, final boolean excludeRemoved) {
```

Comment:
instead of passing the user id why dont you pass in the logged in User details.
fetchAcspMembershipsForUser(User user, boolean excludeRemoved){
return acspMembershipListMapper.daoToDto(
        acspMembersRepository
            .fetchAcspMembersByUserId(userId)
            .filter(acspMembersDao -> excludeRemoved && !acspMembersDao.hasBeenRemoved())
            .toList(),
        User);

}

i think your filter is wrong.
have a look above and see if it fits.

Reply:
I think the filtering is right but the way the code is written it makes it hard to understand - I'll change some stuff to hopefully improve clarity/cognitive complexity.

Reply:
please pass in a user object into the service so that you dont need to use user context.
a method should be independent and should not depend on any global values. Hard to read the code.

Reply:
^ This comment/code is outdated